pkg/cmd/cli/cli.go

```diff
@@ -62,43 +62,76 @@ func NewCommandCLI(name, fullName string) *cobra.Command {
 
 	f := clientcmd.New(cmds.PersistentFlags())
 
-	cmds.AddCommand(cmd.NewCmdLogin(fullName, f, in, out))
-	cmds.AddCommand(cmd.NewCmdLogout("logout", fullName+" logout", fullName+" login", f, in, out))
-	cmds.AddCommand(cmd.NewCmdProject(fullName+" project", f, out))
-	cmds.AddCommand(cmd.NewCmdRequestProject("new-project", fullName+" new-project", fullName+" login", fullName+" project", f, out))
-	cmds.AddCommand(cmd.NewCmdNewApplication(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdStatus(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdNewBuild(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdStartBuild(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdCancelBuild(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdBuildLogs(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDeploy(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdRollback(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdEnv(fullName, f, os.Stdin, out))
-	cmds.AddCommand(cmd.NewCmdExpose(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdGet(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDescribe(fullName, f, out))
-	// Deprecate 'osc apply' with 'osc create' command.
-	cmds.AddCommand(applyToCreate(cmd.NewCmdCreate(fullName, f, out)))
-	cmds.AddCommand(cmd.NewCmdProcess(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdEdit(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdUpdate(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDelete(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdLogs(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdImportImage(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdExec(fullName, f, os.Stdin, out, os.Stderr))
-	cmds.AddCommand(cmd.NewCmdPortForward(fullName, f))
-	cmds.AddCommand(cmd.NewCmdProxy(fullName, f, out))
+	groups := templates.CommandGroups{
+		{
+			Message: "Basic Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdLogin(fullName, f, in, out),
+				cmd.NewCmdRequestProject("new-project", fullName+" new-project", fullName+" login", fullName+" project", f, out),
+				cmd.NewCmdProject(fullName+" project", f, out),
+				cmd.NewCmdNewApplication(fullName, f, out),
+				cmd.NewCmdStatus(fullName, f, out),
+			},
+		},
+		{
+			Message: "Build Code and Deploy Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdStartBuild(fullName, f, out),
+				cmd.NewCmdBuildLogs(fullName, f, out),
+				cmd.NewCmdDeploy(fullName, f, out),
+				cmd.NewCmdRollback(fullName, f, out),
+				cmd.NewCmdNewBuild(fullName, f, out),
+				cmd.NewCmdCancelBuild(fullName, f, out),
+				cmd.NewCmdImportImage(fullName, f, out),
+			},
+		},
+		{
+			Message: "Application Modification Commands:",
```

Comment:
Would you expect it to be used by anyone except admins and people who knew what they were doing?  I.e. "Hey, Steve P, use create"?

> On Jun 6, 2015, at 12:09 PM, Jordan Liggitt <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cli.go:
> 
> > +			},
> > +		},
> > +		{
> > +			Message: "Build Code and Deploy Commands:",
> > +			Commands: []*cobra.Command{
> > +				cmd.NewCmdStartBuild(fullName, f, out),
> > +				cmd.NewCmdBuildLogs(fullName, f, out),
> > +				cmd.NewCmdDeploy(fullName, f, out),
> > +				cmd.NewCmdRollback(fullName, f, out),
> > +				cmd.NewCmdNewBuild(fullName, f, out),
> > +				cmd.NewCmdCancelBuild(fullName, f, out),
> > +				cmd.NewCmdImportImage(fullName, f, out),
> > +			},
> > +		},
> > +		{
> > +			Message: "Application Modification Commands:",
> I'm not sure about the groupings of the things in here and in "advanced commands"... create is advanced?
> 
> —
> Reply to this email directly or view it on GitHub.
>